src/Controls/src/Core/Platform/Windows/ImageConverter.cs

```diff
@@ -8,9 +9,16 @@ public class ImageConverter : Microsoft.UI.Xaml.Data.IValueConverter
 	{
 		public object Convert(object value, Type targetType, object parameter, string language)
 		{
-			return value is ImageSource source
-				? source.ToIconSource(source.FindMauiContext())?.CreateIconElement()
-				: null;
+			ImageSource source = value as ImageSource;
+			if (source == null)
+			{
+				return null;
+			}
+
+			IMauiContext context = source.FindMauiContext(true);
+			IImageSourceServiceProvider imageSourceServiceProvider = context.Services.GetRequiredService<IImageSourceServiceProvider>();
+			IImageSourceService imageSourceService = imageSourceServiceProvider.GetImageSourceService(source);
+			return imageSourceService.GetImageSourceAsync(source).Result;
```

Comment:
Previously we were using this extension method to return the value from here "safely"

https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.UAP/ImageConverter.cs#L12

should we go back to that?
https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Core/Internals/AsyncValue.cs#L95

`.result` is always scary